plugins/modules/ip_share.py

```diff
@@ -0,0 +1,159 @@
+
+import ansible_collections.linode.cloud.plugins.module_utils.doc_fragments.ip_share as ip_share_docs
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
+    LinodeModuleBase,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
+    global_authors,
+    global_requirements,
+)
+from ansible_specdoc.objects import (
+    FieldType,
+    SpecDocMeta,
+    SpecField,
+    SpecReturnValue,
+)
+from linode_api4.objects import Instance
+
+ip_share_spec = {
+    # Disable the default values
+    "label": SpecField(type=FieldType.string, required=False, doc_hide=True),
+    "state": SpecField(type=FieldType.string, required=False, doc_hide=True),
+    "ips": SpecField(
+        type=FieldType.list,
+        required=True,
+        description=[
+            "A list of secondary Linode IPs to share with the primary Linode."
+        ],
+    ),
+    "linode_id": SpecField(
+        type=FieldType.integer,
+        required=True,
+        description=[
+            "The ID of the primary Linode that the addresses will be shared with."
+        ],
+    ),
+}
+
+SPECDOC_META = SpecDocMeta(
+    description=["Manage the Linode shared IPs."],
+    requirements=global_requirements,
+    author=global_authors,
+    options=ip_share_spec,
+    examples=ip_share_docs.specdoc_examples,
+    return_values={
+        "ip_share_stats": SpecReturnValue(
+            description="The Linode IP share info in JSON serialized form",
+            docs_url="https://www.linode.com/docs/api/networking/"
+            + "#ip-addresses-share__response-samples",
+            type=FieldType.dict,
+            sample=ip_share_docs.result_ip_share_stats_samples,
+        )
+    },
+)
+
+
+class IPShareModule(LinodeModuleBase):
+    """Module for configuring Linode shared IPs."""
+
+    def __init__(self) -> None:
+        self.module_arg_spec = SPECDOC_META.ansible_spec
+        self.results = {
+            "changed": False,
+            "actions": [],
+            "linode_id": None,
+            "ips": None,
+        }
+
+        super().__init__(
+            module_arg_spec=self.module_arg_spec,
+        )
+
+    def _share_ip_addresses(self, ips: List[str], linode_id: str) -> None:
+        """
+        Configure shared IPs.
+        """
+        try:
+            self.client.networking.ip_addresses_share(
+                ips=ips,
+                linode=linode_id,
+            )
+        except Exception as exception:
+            self.fail(
+                msg="failed to configure shared ips for linode {0}: {1}".format(
+                    linode_id, exception
+                )
+            )
+
+    # check if the IPs have been shared with the Linode instance
+    def _check_shared_ip_addresses(
+        self, ips: List[str], linode: Instance
+    ) -> bool:
+        linode_shared_ipv4 = [i.address for i in linode.ips.ipv4.shared]
+        linode_public_ipv6 = [i.range for i in linode.ips.ipv6.ranges]
```

Comment:
To ensure the returned IPv6 ranges are only shared, could ensure that `is_bgp` is `True` for each of these ipv6 ranges? You might need to make a few extra network requests to the [IPv6 Range View](https://www.linode.com/docs/api/networking/#ipv6-range-view) endpoint since `is_bgp` seems to be missing from some IPv6 response bodies.

Reply:
Good catch! I added this check to ensure `is_bgp` is True for these ipv6 ranges.